Client/Client.cs

```diff
@@ -183,61 +210,16 @@ private void ReadConfiguration()
             this.azureADServiceHost = Configuration.GetElementValue("AzureADServiceHost");
             this.apiVersion = Configuration.GetElementValue("ApiVersion");
             this.protectedResourcePrincipalId = Configuration.GetElementValue("ProtectedResourcePrincipalId");
+            this.authority = Configuration.GetElementValue("Authority");
         }
 
         /// <summary>
-        /// Adds the required headers to the specified web client.
-        /// </summary>
-        /// <param name="webClient">Web client to add the required headers to.</param>
-        private void AddHeaders(WebClient webClient)
-        {
-            webClient.Headers.Add(Constants.HeaderNameAuthorization, this.GetAccessToken().Result);
-            webClient.Headers.Add(HttpRequestHeader.Accept, "application/json");
-        }
-
-        /// <summary>
-        /// Constructs the URI with the specified entitySet and downloads it with the specified web client.
-        /// </summary>
-        /// <param name="webClient">Web client to be used to download the URI.</param>
-        /// <param name="entitySet">Entity 
-        /// to be used to construct the URI.</param>
-        /// <returns>Byte array containing the downloaded URI.</returns>
-        private byte[] DownloadData(WebClient webClient, string entitySet, string skiptoken)
-        {
-            this.AddHeaders(webClient);
-            string serviceEndPoint = null;
-            if (string.IsNullOrEmpty(skiptoken))
-            {
-                serviceEndPoint = string.Format(
-                @"https://{0}/{1}/{2}/delta",
-                this.azureADServiceHost,
-                this.apiVersion,
-                entitySet);
-            }
-            else
-            {
-                serviceEndPoint = skiptoken;
-            }
-
-            // Log the query string and endpoint.
-            if (this.logger != null)
-            {
-                this.logger.LogDebug("Making call to endpoint : {0}", serviceEndPoint);
-                this.logger.LogDebug("Query Parameters : {0}", LogQueryParameters(webClient.QueryString));
-            }
-
-            return webClient.DownloadData(serviceEndPoint);
-        }
-
-        /// <summary>
-        /// Delegate to invoke the specified operation.
+        /// Returns a valid IAuthenticationProvider object to be used for creating a GraphClient
         /// </summary>
-        /// <param name="operation">Operation to invoke.</param>
-        private void InvokeOperation(Action operation)
+        private IAuthenticationProvider GetAuthorizationProvider(String [] scopes, String authority)
         {
-            // - retry mechanism to be implemented
-            //
-            operation();
+            PublicClientApplication clientApplication = new PublicClientApplication(this.appPrincipalId, authority);
```

Comment:
We should rename appPrincipalId to clientId.  And is appPrinicpalPassword used?  This sample seems to use Interactive flow, not ClientCredentials flow.